File: src/main/java/com/example/the_machine/service/events/RunCreatedEvent.java
package com.example.the_machine.service.events;

import java.util.UUID;
import org.springframework.transaction.event.TransactionalEventListener;

/**
 * Event published when a run is created and ready for execution. Used with
 * {@link TransactionalEventListener} to ensure async execution starts only after the transaction
 * commits.
 */
public record RunCreatedEvent(UUID runId, UUID threadId) {

}

File: src/main/kotlin/com/example/the_machine/service/events/RunCreatedEvent.kt
package com.example.the_machine.service.events

import org.springframework.transaction.event.TransactionalEventListener
import java.util.*

/**
 * Event published when a run is created and ready for execution. Used with
 * [TransactionalEventListener] to ensure async execution starts only after the transaction
 * commits.
 */
data class RunCreatedEvent(
  val runId: UUID,
  val threadId: UUID
)